pkg/gameserverallocations/allocator.go

```diff
@@ -509,11 +510,23 @@ func (c *Allocator) ListenAndAllocate(ctx context.Context, updateWorkerCount int
 				requestCount = 0
 			}
 
+			// FeatureCountsAndLists: Sorting based on GameServerAllocation Priority only for strategy Distributed.
```

Comment:
So another suggestion that may make this easier to manage.

Duplicate `ListenAndAllocate` into a `ListenAndAllocateWithListsAndCounter` (or duplicate the old version, whichever way works).

Then the feature branch when disabled can be to run `ListenAndAllocate` and the feature branch when enabled could be to run `ListenAndAllocateWithListsAndCounter`.

Then you don't have to run as many intricate if/else statements -- and you don't have to worry about breaking existing systems either.

Then you can also add new functions to `allocation_cache.go` that are probably almost exact copies of `ListSortedGameServers` etc, but again, less if/else branching and probably easier to reconcile.

How does that sound?